plugins/circleci/src/state/useBuildWithSteps.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { errorApiRef, useApi } from '@backstage/core';
+import { useContext } from 'react';
+import { circleCIApiRef, GitType } from '../api/index';
+import { AppContext } from '.';
+import { useSettings } from './useSettings';
+
+import { useAsyncPolling } from './useAsyncPolling';
+
+const INTERVAL_AMOUNT = 3000;
+
+export function useBuildWithSteps(buildId: number) {
+  const [settings] = useSettings();
+  const [{ buildsWithSteps }, dispatch] = useContext(AppContext);
+  const api = useApi(circleCIApiRef);
+  const errorApi = useApi(errorApiRef);
+
+  /* eslint-disable-line no-use-before-define */
+  const { isPolling, startPolling, stopPolling } = useAsyncPolling(
```

Comment:
@Rugvip hey! can you take a look? any ideas on how to avoid this?

Reply:
I think you want to remove `isPolling`, and if needed, add it back as a separate ref. I think don't think this works as intended atm anyway, since isPolling is backed by a ref and won't cause rerender when it changes, and since getBuildWithSteps closes over the returned value, you might not always stop when you intended to.

Reply:
Got it. I'll try to find another way.

Reply:
- [ ] Fix functions crossreference